pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/IntersectAllOperatorTest.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.operator;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.pinot.common.datablock.DataBlock;
+import org.apache.pinot.common.utils.DataSchema;
+import org.apache.pinot.query.routing.VirtualServerAddress;
+import org.apache.pinot.query.runtime.blocks.TransferableBlock;
+import org.apache.pinot.query.runtime.blocks.TransferableBlockTestUtils;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+
+public class IntersectAllOperatorTest {
+
+  private AutoCloseable _mocks;
+
+  @Mock
+  private MultiStageOperator _leftOperator;
+
+  @Mock
+  private MultiStageOperator _rightOperator;
+
+  @Mock
+  private VirtualServerAddress _serverAddress;
+
+  @BeforeMethod
+  public void setUp() {
+    _mocks = MockitoAnnotations.openMocks(this);
+    Mockito.when(_serverAddress.toString()).thenReturn(new VirtualServerAddress("mock", 80, 0).toString());
+  }
+
+  @AfterMethod
+  public void tearDown()
+      throws Exception {
+    _mocks.close();
+  }
+
+  @Test
+  public void testIntersectAllOperator() {
+    DataSchema schema = new DataSchema(new String[]{"int_col"},
+        new DataSchema.ColumnDataType[]{DataSchema.ColumnDataType.INT});
+
+    Mockito.when(_leftOperator.nextBlock())
+        .thenReturn(OperatorTestUtil.block(schema, new Object[]{1}, new Object[]{2}, new Object[]{3}))
+        .thenReturn(TransferableBlockTestUtils.getEndOfStreamTransferableBlock(0));
+    Mockito.when(_rightOperator.nextBlock()).thenReturn(
+            OperatorTestUtil.block(schema, new Object[]{1}, new Object[]{2}, new Object[]{4}))
+        .thenReturn(TransferableBlockTestUtils.getEndOfStreamTransferableBlock(0));
```

Comment:
Do you think we could create an actual operator that receives a list of blocks and returns that? It would be great to do not use mocks and simplify these tests.

It is not something I think we should do in this PR, but I want just to open that discussion for the future.

Something like:
```java
_leftOperator = new BlockListOperator.builder()
     .andThen(block1)
     .andThen(block2)
     .endWith(eosBlock(stats)) // or errorBlock(errors)
```

That is not specially better than mocks, but has the advantage of not returning null as soon as we call other method and it would mean there is a single place to apply changes in future refactors.

Reply:
> but has the advantage of not returning null as soon as we call other method

We could use a spy or partial mock, but if we're able to use the actual operator without much setup overhead, this definitely makes more sense.

> it would mean there is a single place to apply changes in future refactors

I didn't fully follow this part though.

> It is not something I think we should do in this PR, but I want just to open that discussion for the future.

I can make this change in a follow-up PR for both these new tests as well as the existing tests for the other operators.